python/ray/data/_internal/logical/operators/map_operator.py

```diff
@@ -22,44 +23,99 @@
     from typing_extensions import Literal
 
 
-class MapBatches(LogicalOperator):
-    """Logical operator for map_batches."""
+class BaseMapLike(LogicalOperator):
```

Comment:
This name looks strange, maybe LogicalMap or AbstractMap or MapInterface?

Reply:
@jianoaix - sure, changed to `AbstractMap`.